consensus/validation_fuzz.go

```diff
@@ -0,0 +1,67 @@
+// +build gofuzz
+
+package consensus
+
+import (
+	"crypto/ed25519"
+	"go.sia.tech/core/types"
+)
+
+func Fuzz(data []byte) int {
+	d := types.NewBufDecoder(data)
+
+	var txn types.Transaction
+	txn.DecodeFrom(d)
+	if err := d.Err(); err != nil {
+		return -1
+	}
+	txn.FileContractResolutions = nil
+
+	var setupTxn types.Transaction
+	for _, sci := range txn.SiacoinInputs {
+		setupTxn.SiacoinOutputs = append(setupTxn.SiacoinOutputs, sci.Parent.SiacoinOutput)
+	}
+	for _, sfi := range txn.SiafundInputs {
+		setupTxn.SiafundOutputs = append(setupTxn.SiafundOutputs, sfi.Parent.SiafundOutput)
+	}
+	for _, fcr := range txn.FileContractRevisions {
+		setupTxn.FileContracts = append(setupTxn.FileContracts, fcr.Parent.FileContract)
```

Comment:
these outputs will need to use `pubkey`:
```go
	for i, sci := range txn.SiacoinInputs {
		sci[i].Parent.Address = types.StandardAddress(pubkey)
 		setupTxn.SiacoinOutputs = append(setupTxn.SiacoinOutputs, sci.Parent.SiacoinOutput)
 	}
```

Reply:
now they need to use `anyoneCanSpendAddress`:
```go
	for i, sci := range txn.SiacoinInputs {
		sci[i].Parent.Address = anyoneCanSpendAddress
 		setupTxn.SiacoinOutputs = append(setupTxn.SiacoinOutputs, sci.Parent.SiacoinOutput)
 	}
```

Basically, we need to overwrite the addresses *before* we call `ApplyBlock`. `ApplyBlock` is going to create a new accumulator that contains the outputs in `setupTxn`; even if you modify `NewSiacoinElements` afterward, the accumulator itself is unaffected (it's just a bunch of hashes). `ValidateTransaction` will use the transaction's Merkle proofs to determine whether each element is in the accumulator or not, so if you modify the elements *after* the accumulator is constructed, the Merkle proofs will become invalid and `ValidateTransaction` will report that the elements aren't in the accumulator.

Reply:
But sci is an element of txn.SiacoinInputs and cannot be indexed

Reply:
oops, sorry for the confusion there. Should be `txn.SiacoinInputs[i]`